src/components/OcRecipient/OcRecipient.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+  <span class="oc-recipient">
+    <template v-if="recipient.hasAvatar">
+      <oc-spinner
+        v-if="recipient.isLoadingAvatar"
+        key="recipient-avatar-spinner"
+        size="small"
+        :aria-label="$gettext('Loading avatar')"
+        data-testid="recipient-avatar-spinner"
+      />
+      <oc-avatar
+        v-else
+        :key="recipient.avatar || recipient.name"
+        data-testid="recipient-avatar"
+        class="oc-recipient-avatar"
+        :src="recipient.avatar"
+        :user-name="recipient.name"
+        :width="16.8"
+      />
+    </template>
+    <oc-icon
+      v-else
+      class="oc-recipient-icon"
+      size="small"
+      :name="recipient.icon.name"
+      :accessible-label="recipient.icon.label"
+      data-testid="recipient-icon"
+    />
+    <p class="oc-recipient-name" data-testid="recipient-name" v-text="recipient.name" />
+    <!-- @slot Append content (actions, additional info, etc.)  -->
+    <slot name="append" />
+  </span>
+</template>
+
+<script>
+import OcAvatar from "../avatars/OcAvatar.vue"
+import OcIcon from "../OcIcon.vue"
+import OcSpinner from "../OcSpinner.vue"
+
+export default {
+  name: "OcRecipient",
+  status: "review",
+  release: "unreleased",
+
+  components: { OcAvatar, OcIcon, OcSpinner },
+
+  props: {
+    /**
+     * Recipient object containing name (required), icon and avatar
+     */
+    recipient: {
+      type: Object,
+      required: true,
+      validator: recipient => {
+        if (!Object.prototype.hasOwnProperty.call(recipient, "name")) {
+          throw new Error("Recipient name is not defined")
+        }
+
+        if (typeof recipient.name !== "string") {
+          throw new Error("Recipient name is not a string")
+        }
+
+        if (recipient.name.length < 1) {
+          throw new Error("Recipient name is empty")
+        }
+
```

Comment:
Please add validation about if an `icon` key is present in the recipient, that it requires to have the sub keys `recipient.icon.name` and `recipient.icon.label`.